src/skin/legacyskinparser.cpp

```diff
@@ -1084,6 +1089,30 @@ QWidget* LegacySkinParser::parseCoverArt(QDomElement node) {
     return pCoverArt;
 }
 
+void LegacySkinParser::parseSingletonDefinition(QDomElement node) {
+    // The actual singleton definition is identical to any other WWidgetGroup.
+    QWidget* child_group = parseWidgetGroup(node);
+    m_pContext->getSingletonMap()->defineSingleton(child_group->objectName(),
+                                                   child_group);
+    child_group->hide();
+}
+
+QWidget* LegacySkinParser::parseSingletonContainer(QDomElement node) {
```

Comment:
ah, so I may have found the compelling reason to do it this way.  I realized that I also have the spinnies in a widgetstack, so there are 2 spinnies for each deck.  In trying to make them singletons, I needed to be able to specify a different size for each spinny inside the SingletonContainer tag.  If we do standard widget parsing, we get the Size tag for free.

I also moved ObjectName into a tag so that variables can be used (variable parsing doesn't work in an attribute)